cow_instrument/InstrumentTree.h

```diff
@@ -17,30 +16,38 @@ class Detector;
  */
 class InstrumentTree {
 public:
-  InstrumentTree(std::unique_ptr<const Node> &&root, size_t nDetectors);
-
-  InstrumentTree(const InstrumentTree& other);
-
-  std::unique_ptr<NodeIterator> iterator() const;
+  InstrumentTree(std::vector<Node> &&nodes, size_t nDetectors);
+  InstrumentTree(const InstrumentTree& original) = default;
```

Comment:
Defining the copy constructor gets rid of the move constructor and assignment operators, is this intentional?